benchmarks/vendor_allocator_benchmarks.cpp

```diff
@@ -0,0 +1,183 @@
+//////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2016-20, Lawrence Livermore National Security, LLC and Umpire
+// project contributors. See the COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (MIT)
+//////////////////////////////////////////////////////////////////////////////
+#include "benchmark/benchmark.h"
+
+#include "umpire/Allocator.hpp"
+#include "umpire/ResourceManager.hpp"
+
+#include "umpire/alloc/MallocAllocator.hpp"
+#if defined(UMPIRE_ENABLE_CUDA)
+#include "umpire/alloc/CudaMallocAllocator.hpp"
+#include "umpire/alloc/CudaMallocManagedAllocator.hpp"
+#include "umpire/alloc/CudaPinnedAllocator.hpp"
+#endif
+#if defined(UMPIRE_ENABLE_HIP)
+#include "umpire/alloc/HipMallocAllocator.hpp"
+#include "umpire/alloc/HipMallocManagedAllocator.hpp"
+#include "umpire/alloc/HipPinnedAllocator.hpp"
+#endif
+
+static const int RangeLow{1048576}; //2^20 or ~1MB
```

Comment:
Can the smallest allocation be smaller than 1 meg, maybe something like 1 kb.

Reply:
I could definitely do that, but as I test it on lassen, it takes longer than 2 hours to complete a run of the benchmark when I  have ENABLE_CUDA on. If it's just the host, I can do it within the timeframe of the default lalloc job. (For the job that I ran, it ranged from 1024 to 268435456 with a range multiplier of 4 as before) Is that ok?

Reply:
[cuda-10-1_vendor_allocator_benchmarks_1kBto1GB.txt](https://github.com/LLNL/Umpire/files/5921037/cuda-10-1_vendor_allocator_benchmarks_1kBto1GB.txt)

(For reference)